### DISCUSSION TITLE
Your background

### DISCUSSION THREAD

**Discussion by Memes1125** [AveYo/ShowMMR#19] [Ideas]
How do you like the idea of adding a background selection function. For example, I want beautiful girls to stand in my background and with the help of this mod I could implement it. Thanks, I think it's a great idea. Sorry if it's not clear, I used a translator. 
![image](https://github.com/AveYo/ShowMMR/assets/73619063/3a35a979-79c7-4a81-98f3-24c5f833d63e)

**Comment by AveYo** [comment]
It's doable, but the background would have to be precompiled to texture, can't use a png directly for example.
In a [previous version](https://github.com/AveYo/ShowMMR/commit/3bbbc9a2955035fe527110a855a6988c3cb99bfd) I did replace the Bali Major background - and immediately after, Valve fixed theirs so they are watching ;)
I guess I can add another variant where you can add your own background as png, and then the script will resource-compile it to vtex_c automatically. No time for it atm.

**Comment by Memes1125** [comment]
If it's not difficult, I would be happy with this script and an explanation of how to do everything. So that the mod does not conflict with the background. 😊😊

**Comment by AveYo** [comment]
explore [source/custom_background](https://github.com/AveYo/ShowMMR/tree/main/source/custom_background) variant for a live example  

currently using the meme background, with original size on frontpage or fullscreen elsewhere
you can replace `source/custom_background/dota/background.png` with your own
to have it always be fullscreen delete:
```css
  width: 800px;
  height: 450px;
  margin-bottom: 300px;
```
in source/custom_background/dota/panorama/styles/dashboard_background_default.css `.OnHomePage #Custom{ ... }`

next update I will change it a bit and make the image always fullscreen

**Reply by Memes1125** [reply]
Good afternoon, I tried to put my background, I create a VPK file through dota_mod_builder.bat, I deleted in styles what you said as a result, my picture is not put. But when I take your release, then your picture appears. Maybe I'm doing something wrong?

**Reply by AveYo** [reply]
Just wait for an update, will provide a simpler way and remove the meme stuff that was just temporary

**Reply by Memes1125** [reply]
Well I will wait ☺☺

**Reply by Memes1125** [reply]
Does the latest release imply replacing the background with your own?

**Reply by Memes1125** [reply]
![image](https://github.com/AveYo/ShowMMR/assets/73619063/f4711b8a-a666-4acc-ad10-0ab22cbcea4b)
Valve may have broken your mod for the mmr show
